### PR TITLE
feat: add interpreter_version_info to py_runtime

### PR DESCRIPTION
**Pull request by mattem**
Adds an `interpreter_version_info` attribute to the `py_runtime` and associated provider that maps to the `sys.version_info` values. This allows the version of the interpreter to be known statically, which can be useful for rule sets that depend on the interpreter, and need to build environments / pathing that contain version info (virtualenvs for example).

### CODE REVIEW

File: python/private/common/py_runtime_rule.bzl [hunk 1]
```diff
@@ -61,6 +61,10 @@ def _py_runtime_impl(ctx):
 
     python_version = ctx.attr.python_version
 
+    interpreter_version_info = ctx.attr.interpreter_version_info
+    if interpreter_version_info and len(interpreter_version_info) < 2:
```
Comment (mattem) [inline python/private/common/py_runtime_rule.bzl:65]: We can expand this to include micro if we want

File: python/private/common/providers.bzl [hunk 2]
```diff
@@ -136,6 +138,11 @@ the same conventions as the standard CPython interpreter.
             "filesystem path to the interpreter on the target platform. " +
             "Otherwise, this is `None`."
         ),
+        "interpreter_version_info": (
```
Comment (mattem) [inline python/private/common/providers.bzl:141]: Let the bikeshedding on the name commence...

File: python/private/common/py_runtime_rule.bzl [hunk 3]
```diff
@@ -196,6 +206,14 @@ platform runtime this attribute must not be set.
 For a platform runtime, this is the absolute path of a Python interpreter on
 the target platform. For an in-build runtime this attribute must not be set.
 """),
+        "interpreter_version_info": attr.string_list(
+            doc = """
+This is the version of the interpreter that this runtime provides.
+The list can contain the five components of the version number: major, minor, micro, releaselevel, and serial.
+This should match the format given by `sys.version_info`, however for simplicity the micro, releaselevel and serial values may be omitted.
```
Comment (f0rmiga) [inline python/private/common/py_runtime_rule.bzl:213]:
```suggestion
This should match the format given by `sys.version_info`, however for simplicity, the micro, releaselevel and serial values may be omitted.
```

**Review by f0rmiga — APPROVED** [review]
I'll let other folks bikeshed if they want.

File: python/private/common/providers.bzl (hunk 2 above)
Comment (rickeylev) [reply python/private/common/providers.bzl:141]: I'm OK with `interpreter_version_info`, as that is what it's representing. `python_version_info` or `version_info` would also be OK IMHO.

**Review by rickeylev — CHANGES_REQUESTED** [review]
My main comment is to use a dict/struct instead of list. This makes it easier to know which parts are check for the relevant pieces. Plus we can stash extra implementation-specific parts in there if necessary.

i.e. accept a attr.string_dict on the rule and and convert it to a struct() for the provider. Convert particular keys to their appropriate types (e.g. major/minor/patch to ints, similar to sys.version_info)

Please add tests. You can add them to //tests/py_runtime

**Comment by mattem** [comment]
@rickeylev Can I have another pass on this please?

**Comment by rickeylev** [comment]
Looks pretty good! Thank you for adding tests.

Can you please add an entry to CHANGELOG.md? Then it's ready to merge.

**Review by rickeylev — APPROVED** [review]

**Pull request merged**